python/lsst/dax/distribution/DataIngest.py

```diff
@@ -0,0 +1,297 @@
+        status : int
+            Status code from the put operation.
+        r_json : json or None
+            json data from the put operation if status was 200.
+        """
+        cmd = 'ingest/trans/' + str(transaction_id) + '?abort='
+        if abort:
+            cmd += '1'
+        else:
+            cmd += '0'
+        success, status, r_json = self._putToIngest(cmd, {'auth_key':self._auth_key})
+        if not success:
+            print("ERROR ending transaction id=", transaction_id, "abort=", abort, "status=", status,
+                  "r_json=", r_json)
+        return success, status, r_json
+
+    def getChunkTargetAddr(self, transaction_id, chunk_id):
+        """ Get the host and port number of the ingest worker for this chunk.
+
+        Parameters
+        ----------
+        transaction_id : int
+            Ingest transaction id number.
+        chunk_id : int
+            Chunk id number.
+
+        Return
+        ------
+        host : str
+            Host name of the ingest worker.
+        port : int
+            Port number of the ingest worker.
+        """
+        cmd = 'ingest/chunk'
+        jdata = {"transaction_id":transaction_id,"chunk":chunk_id,"auth_key":self._auth_key}
+        success, r_json = self._postToIngest(cmd, jdata)
+        if not success:
+            print("ERROR failed to get chunk target address", jdata, " r_json=", r_json)
+            raise RuntimeError('Transaction ' + transaction_id
+                               + ' failed to get target address for chunk ' + chunk_id)
+        host = r_json['location']['host']
+        port = r_json['location']['port']
+        return host, port
+
+    def sendChunkToTarget(self, host, port, transaction_id, table, f_path):
+        """ Send the file to the ingest worker.
+
+        Parameters
+        ----------
+        host : str
+            Host name for the ingest worker.
+        port : int
+            Port number for the ingest worker.
+        transation_id : int
+            Ingest transaction id number.
+        table : str
+            The name of the table that f_path data should be added to.
+        f_path : str
+            Full path to the file to ingest.
+
+        Note
+        ----
+        This calls the external program 'qserv-replica-file-ingest' to
+        actually send the file.
+        """
+        cmd = ('qserv-replica-file-ingest FILE ' + host + ' ' + str(port) + ' '
+            + str(transaction_id) + ' ' + table + ' P ' + f_path + ' --verbose --columns-separator=TAB')
+        print("cmd=", cmd)
+        process = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        out_str = str(process.stdout)
+        if process.returncode != 0:
+            print("ERROR sendChunkToTarget cmd=", cmd, "out=", out_str)
+            raise RuntimeError("ERROR sendChunkToTarget cmd=" + cmd + " out=" + str(out_str))
+        return process.returncode, out_str
```

Comment:
It makes no sense to return the completion code from this method as it would always be 0. Otherwise you will throw exception `RuntimeError`.